txp/Pos/Txp/Core/Types.hs

```diff
@@ -289,58 +256,32 @@ instance NFData TxProof
 -- with different number of transactions and witnesses.
 data TxPayload = UnsafeTxPayload
     { -- | Transactions are the main payload.
-      _txpTxs           :: !(MerkleTree Tx)
+      _txpTxs       :: !(MerkleTree Tx)
     , -- | Witnesses for each transaction. The length of this field is
       -- checked during deserialisation; we can't put witnesses into the same
       -- Merkle tree with transactions, as the whole point of SegWit is to
       -- separate transactions and witnesses.
       --
       -- TODO: should they be put into a separate Merkle tree or left as
       -- a list?
-      _txpWitnesses     :: ![TxWitness]
-    , -- | Stake distribution for each transaction. Follow-the-satoshi will
-      -- use this information to decide who “owns” which satoshi, even though
-      -- in reality we don't know it for e.g. P2SH transactions. So, one of
-      -- the reasons we need distributions is that we want (potential)
-      -- receivers of P2SH funds to count as stakeholders, though there are
-      -- also other things distributions are useful for.
-      _txpDistributions :: ![TxDistribution]
+      _txpWitnesses :: ![TxWitness]
     } deriving (Show, Eq, Generic)
 
 instance NFData TxPayload
 
 makeLenses ''TxPayload
 
--- | Smart constructor of 'TxPayload' which can fail if some
--- invariants are violated.
+-- | Smart constructor of 'TxPayload' which ensures that invariants of 'TxPayload' hold.
 --
--- Currently there are two invariants:
--- • number of txs must be same as number of witnesses and
---   number of distributions;
--- • each distribution must have the same number of elements as
---   number of outputs in corresponding transaction.
-mkTxPayload
-    :: (Bi Tx, MonadFail m)
-    => [TxAux] -> m TxPayload
+-- Currently there is only one invariant:
+-- • number of txs must be same as number of witnesses.
```

Comment:
The invariant seems to be correct by construction (each tx is paired with a witness), so this comment isn't needed.

Reply:
That's the invariant of `TxPayload` type (and that's written in the comment) and it doesn't hold by construction. This constructor enforces it.